src/app/view/directives/dot-maxlength/dot-maxlength.directive.ts

```diff
@@ -24,12 +24,48 @@ export class DotMaxlengthDirective {
      */
     @HostListener('paste', ['$event'])
     @HostListener('keypress', ['$event'])
-    eventHandler() {
-        setTimeout(() => {
-            this.el.nativeElement.textContent = this.el.nativeElement.textContent.slice(
-                0,
-                this._maxLength
-            );
-        }, 0);
+    eventHandler(event: Event) {
+        if (
+            this.el.nativeElement.textContent.length < this._maxLength ||
```

Comment:
This smells, first you do:

this.el.nativeElement.textContent.length < this._maxLength

and then inside yo do the opposite: this.el.nativeElement.textContent.length > this._maxLength

are you sure about this?

Reply:
When the event fires, the textContent have the initial value, with the timeout the value is updated. example, on a key press event of "B" the before the timeout value will be "A", after will be "AB".